### PR TITLE
Improve Btrfs subvols management

### PR DESCRIPTION
**Pull request by joseivanlopez**
## Problem

https://trello.com/c/OQGxR8Eo/2077-5-partitioner-more-integrated-subvolumes-management

## Testing

- *Added a new unit test*
- *Tested manually*

## Screenshots

*If the fix affects the UI attach some screenshots here.*

### CODE REVIEW

File: src/lib/y2storage/filesystems/blk_filesystem.rb [hunk 1]
```diff
@@ -118,6 +118,18 @@ def plain_blk_devices
         blk_devices.map(&:plain_device)
       end
 
+      def name
+        textdomain "storage"
+
+        format(
+          # TRANSLATORS: %{fs_type} is replaced by the filesystem type (e.g., BtrFS), and %{device_name}
```
Comment (mvidner) [inline src/lib/y2storage/filesystems/blk_filesystem.rb:125]: yes, but this still does not tell the translator the *context* of the message that may be useful for the translation

File: src/lib/y2partitioner/widgets/device_table_entry.rb [hunk 2]
```diff
@@ -38,6 +38,8 @@ class DeviceTableEntry
       # @return [Array<DeviceTableEntry>]
       attr_reader :children
 
+      attr_reader :open
```
Comment (mvidner) [inline src/lib/y2partitioner/widgets/device_table_entry.rb:41]:
```suggestion
      # @return [Boolean]
      attr_reader :open
```

File: src/lib/y2partitioner/actions/add_btrfs_subvolume.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    class AddBtrfsSubvolume < Base
+      def initialize(filesystem)
+        super()
+
+        @controller = Controllers::BtrfsSubvolume.new(filesystem: filesystem)
+      end
+
+      private
+
+      # @return [Controllers::Bcache]
```
Comment (mvidner) [inline src/lib/y2partitioner/actions/add_btrfs_subvolume.rb:37]: really Bcache?

File: src/lib/y2partitioner/actions/controllers/btrfs_subvolume.rb [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      # a Btrfs filesystem
+      class BtrfsSubvolume < Base
+        # @return [Y2Storage::Filesystems::Btrfs]
+        attr_reader :filesystem
+
+        attr_reader :subvolume
+
+        attr_accessor :subvolume_path
+
+        attr_accessor :subvolume_nocow
```
Comment (mvidner) [inline src/lib/y2partitioner/actions/controllers/btrfs_subvolume.rb:35]: Please add docs with the correct types, they are important for complex code like this

File: src/lib/y2partitioner/actions/edit_btrfs_subvolume.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        @controller = Controllers::BtrfsSubvolume.new(subvolume: subvolume)
+      end
+
+      private
+
+      # @return [Controllers::Bcache]
+      attr_reader :controller
+
+      # Opens a dialog to create a bcache
```
Comment (mvidner) [inline src/lib/y2partitioner/actions/edit_btrfs_subvolume.rb:40]: again many mentions of bcache, it looks like copy-paste leftovers

File: src/lib/y2partitioner/actions/controllers/btrfs_subvolume.rb (hunk 4 above)
Comment (joseivanlopez) [reply src/lib/y2partitioner/actions/controllers/btrfs_subvolume.rb:35]: This is still a WIP. Please, do not review it yet.

**Pull request closed without merging**